solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -852,6 +815,68 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         return outputAmount.add(_reserveAmount.mul(compN).div(compD));
     }
 
+    /**
+      * @dev returns the amount of tokens required for liquidation
+      *
+      * @param _poolToken       pool token
+      * @param _reserveToken    reserve token
+      * @param _poolAmount      pool token amount when the liquidity was added
+      * @param _targetAmount    target token amount received in return
+      * @param _isBaseAmount    true if the required amount is of the based token
+      * @return amount of tokens required for liquidation
+    */
+    function liquidationAmount(
+        IDSToken _poolToken,
+        IERC20Token _reserveToken,
+        uint256 _poolAmount,
+        uint256 _targetAmount,
+        bool _isBaseAmount)
+        internal view returns (uint256)
+    {
+        // calculate the amount of pool tokens required for liquidation
+        // note that the amount is doubled since it's not possible to liquidate one reserve only
+        Fraction memory poolRate = poolTokenRate(_poolToken, _reserveToken);
+        uint256 poolAmount = _targetAmount.mul(poolRate.d).div(poolRate.n / 2);
+
+        // limit the amount of pool tokens by the amount that the system holds
+        uint256 systemBalance = store.systemBalance(_poolToken).add(_poolAmount);
+        poolAmount = poolAmount > systemBalance ? systemBalance : poolAmount;
+
+        if (_isBaseAmount) {
+            // calculate the base token amount received by liquidating the pool tokens
+            // note that the amount is divided by 2 since the pool amount represents both reserves
+            return poolAmount.mul(poolRate.n / 2).div(poolRate.d);
```

Comment:
seems like the function return base token amount in this case and not pool token amount while it does return pool token amount in the last line (and that's what the description says).
I think we should aim for more consistency here, 2 different return values that also represent different meanings is a bit too much.

Reply:
These are not two different return values.
It's the same return value depicted in different units.
The input argument `_isBaseAmount` indicates whether the return value should be in base token units or not (in which case, they should be in pool token units).

Reply:
it's highly unconventional nonetheless, and even got me confused